src/starbase/setup.ts

```diff
@@ -2,18 +2,77 @@ import { Clone, Reference, Repository } from 'nodegit';
 import { StarbaseConfig, StarbaseIntegration } from './types';
 import { isDirectoryPresent } from '@jupiterone/integration-sdk-runtime';
 import { executeWithLogging } from './process';
+import snakecase from 'lodash.snakecase';
+import mapkeys from 'lodash.mapkeys';
+import Ajv, { Schema } from 'ajv';
+
+// Set up Ajv to return all errors instead of just the first and
+// allow for the 'mask' keyword used in our instanceConfigFields.
+const ajv = new Ajv({ allErrors: true });
+ajv.addKeyword('mask');
 
 async function setupStarbase(config: StarbaseConfig) {
+  let configErrorCount = 0;
   for (const integration of config.integrations) {
     await setupIntegration(integration);
+    if (!(await checkInstanceConfigFields(integration))) {
+      configErrorCount++;
+    }
     await installIntegrationDependencies(integration.directory);
   }
+  if (configErrorCount > 0) {
+    console.error(
+      `ERROR:  Configurations for ${configErrorCount} integration(s) failed validation tests.  Please correct above error messages and retry.`,
+    );
+  }
 }
 
 async function installIntegrationDependencies(directory: string) {
   return executeWithLogging(`yarn --cwd ${directory} install`);
 }
 
+/**
+ * After we have cloned the repository, pull down information on the required
+ * instanceConfigFields and perform additional checks.  Returns false if the
+ * config fields weren't successfully validated.
+ */
+async function checkInstanceConfigFields(
+  integration: StarbaseIntegration,
+): Promise<boolean> {
+  let checkSuccessful = true;
+  await import(`../.${integration.directory}/src/index`).then(
```

Comment:
Consider using `path.resolve()` here to support Windows?

```ts
const integrationConfigModule = await import(path.resolve(`../.${integration.directory}/src/index`));
```

Reply:
Verified that this line is working as intended on a Windows machine.